pilvytis/api.go

```diff
@@ -223,6 +234,135 @@ func (a *API) GetMystExchangeRateFor(curr string) (float64, error) {
 	return rate, nil
 }
 
+// GatewaysResponse holds data about payment gateways.
+type GatewaysResponse struct {
+	Name         string              `json:"name"`
+	OrderOptions PaymentOrderOptions `json:"order_options"`
+	Currencies   []string            `json:"currencies"`
+}
+
+// GetPaymentGateways returns a slice of supported gateways.
+func (a *API) GetPaymentGateways() ([]GatewaysResponse, error) {
+	req, err := requests.NewGetRequest(a.url, "api/v2/payment/gateways", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var resp []GatewaysResponse
+	return resp, a.sendRequestAndParseResp(req, &resp)
+}
+
+// PaymentOrderResponse is a response for a payment order.
+type PaymentOrderResponse struct {
+	ID     string             `json:"id"`
+	Status PaymentOrderStatus `json:"status"`
+
+	Identity       string `json:"identity"`
+	ChainID        int64  `json:"chain_id"`
+	ChannelAddress string `json:"channel_address"`
+
+	GatewayName string `json:"gateway_name"`
+
+	ReceiveMYST float64 `json:"receive_myst"`
```

Comment:
I think we are asking for a problem with floats for monetary values.

Reply:
Can't argue with that. I am more or less copying the old implementation. Thing is a person can request 10.1231231 MYST and that is sort of valid. I guess we could go with `string` or `decimal` of some sort.

Reply:
Well, if it already works, let's leave it as is.